src/transaction.ts

```diff
@@ -1253,15 +1261,15 @@ export class Transaction extends Dml {
 
   batchUpdate(
     queries: Array<string | Statement>,
-    gaxOptions?: CallOptions
+    options?: BatchUpdateOptions
```

Comment:
Because of changes like this, we may actually have to make the type changes separately in time for the breaking change release we'll do in early January. Do you think that would be ok?

Reply:
Yeah, I'll extract this change to a separate PR.

Reply:
Following @JustinBeckwith's suggestion in https://github.com/googleapis/nodejs-spanner/pull/1278#discussion_r540352588, we could also change this parameter (and others) to `BatchUpdateOptions | CallOptions`. The definition of this and the other methods would then look something like this:

```typescript
  batchUpdate(
    queries: Array<string | Statement>,
    optionsOrCallback?: BatchUpdateOptions | CallOptions | BatchUpdateCallback,
    cb?: BatchUpdateCallback
  ): Promise<BatchUpdateResponse> | void {
    const options =
      typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
    const callback =
      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
    const gaxOpts =
      'gaxOptions' in options
        ? (options as BatchUpdateOptions).gaxOptions
        : options;

  ...

  }
```

(@JustinBeckwith Please feel free to correct me if I misunderstood your comment)

Reply:
@JustinBeckwith Friendly ping on this. Would you mind giving us your input/opinion on the above?

Reply:
I tried this out for CommitStats. It works to avoid the breaking change. But there is a caveat that @AVaksman pointed at which is that if options doesn't have `gaxOptions` then we pass the `options` as a whole even if it is not a `CallOptions`.

Is this ok?

I tried to use a [Discriminating Union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) but then we need the user to specify the descriminator field, which would end up being a breaking change anyway.

Reply:
I assume there is no other way to do this than the idea you suggested @olavloite . I'll be going ahead with this implementation for CommitStats. So if you could do the same in this PR, then that works.

Reply:
Yeah - we really want to avoid breaking changes in these library unless it's unreasonable to avoid.  We're starting to evaluate the possibility of backporting fixes to prior major releases, so the cost of these is going to go waaaaay up.  I would recommend avoiding the break if you can.

Reply:
I've changed the different options arguments to `BatchUpdateOptions | CallOptions` (and similar) and added type checking where necessary. That should make this a non-breaking change.